customer-support-endpoint/src/main/kotlin/org/ostelco/prime/support/resources/HoustonResources.kt

```diff
@@ -464,3 +468,59 @@ class CustomerResource {
         }
     }
 }
+
+/**
+ * Resource used to provision new sim profiles.
+ */
+@Path("/support/simprofile")
+
+class SimProfilesResource {
+    private val logger by getLogger()
+    private val storage by lazy { getResource<AdminDataSource>() }
+
+    @EnableTracing
+    @POST
+    @Path("{id}")
+    @Produces(MediaType.APPLICATION_JSON)
+    fun provisionSimProfile(@Auth token: AccessTokenPrincipal?,
+                            @NotNull
+                            @PathParam("id")
+                            id: String,
+                            @QueryParam("regionCode")
+                            regionCode: String,
+                            @QueryParam("profileType")
+                            profileType: String,
+                            @QueryParam("alias")
+                            alias: String): Response =
+            if (token == null) {
+                Response.status(Response.Status.UNAUTHORIZED)
+            } else {
+                logger.info("${token.name} Creating new SIM profile in region $regionCode & profileType $profileType for customerId: $id")
+                provisionSimProfile(
+                        customerId =id,
+                        regionCode = regionCode,
+                        profileType = profileType,
+                        alias = alias)
+                        .responseBuilder(success = Response.Status.CREATED)
+            }.build()
+
+    private fun provisionSimProfile(customerId: String, regionCode: String, profileType: String, alias: String): Either<ApiError, SimProfile> {
```

Comment:
This code has deep indentation and complex control flow.  In my opinion it should be broken up into multiple locally defined functions that are composed in a simple manner so that it can be read at a glance what the method does. THat's not possible as it is written now.

Reply:
I think that if we can avoid the `if (token == null) {` part, then that could help. Means that the `token` check should be moved to a filter or something (as suggested by @la3lma earlier). Should be done in a separate PR though and not in this, as the same pattern is used in all external APIs.